pypeit/reduce.py

```diff
@@ -609,6 +653,61 @@ def local_skysub_extract(self, global_sky, sobjs,
 
         return None, None, None, None, None
 
+    def load_skyregions(self, skymask_init):
+        """
+        Load or generate the sky regions
+
+        Parameters
+        ----------
+        skymask_init :  numpy.ndarray
+            A boolean array of sky pixels (True is pixel is a sky region)
+
+        Returns
+        -------
+        skymask_init :  numpy.ndarray
+            A boolean array of sky pixels (True is pixel is a sky region)
+        usersky : bool
+            If the user has defined the sky, set this variable to True (otherwise False).
+        """
+        usersky = False
+        if self.par['reduce']['skysub']['load_mask']:
+            # Check if a master Sky Regions file exists for this science frame
+            file_base = os.path.basename(self.sciImg.files[0])
+            prefix = os.path.splitext(file_base)
+            if prefix[1] == ".gz":
+                sciName = os.path.splitext(prefix[0])[0]
+            else:
+                sciName = prefix[0]
+
+            # Setup the master frame name
+            master_dir = self.caliBrate.master_dir
+            master_key = self.caliBrate.fitstbl.master_key(0, det=self.det) + "_" + sciName
+
+            regfile = masterframe.construct_file_name(buildimage.SkyRegions,
+                                                      master_key=master_key,
+                                                      master_dir=master_dir)
+            # Check if a file exists
+            if os.path.exists(regfile):
+                msgs.info("Loading SkyRegions file for: {0:s} --".format(sciName) + msgs.newline() + regfile)
+                skymask_init = fits.getdata(regfile).astype(np.bool)
```

Comment:
Shouldn't the read just mimic how we read other `PypeItImages` ?

Reply:
Yeah... that was a total hack by me... I've now implemented this as a `PypeItImage` - thanks for the suggestion!